Include/pythread.h

```diff
@@ -42,7 +42,8 @@ PyAPI_FUNC(unsigned long) PyThread_get_thread_ident(void);
 #if (defined(__APPLE__) || defined(__linux__) || defined(_WIN32) \
      || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) \
      || defined(__OpenBSD__) || defined(__NetBSD__) \
-     || defined(__DragonFly__) || defined(_AIX))
+     || defined(__DragonFly__) || defined(_AIX)) \
+     || defined(__sun__) && SIZEOF_LONG >= 8
```

Comment:
```suggestion
     || defined(__DragonFly__) || defined(_AIX) \
     || (defined(__sun__) && SIZEOF_LONG >= 8))
```

Let's put it inside the group